### DISCUSSION TITLE
Sending decoded JSON data over TCP

### DISCUSSION THREAD

**Discussion by laleft** [jvde-github/AIS-catcher#273] [Q&A] [answered]
Hello everyone. Is there any way I can send the decoded data in JSON format over TCP, as shown in the output with the -o 5 option? My idea is to open a socket from Node.js and read the JSON from there. Thank you!

**Comment by jvde-github** [comment]
Hello, only the -o 3 equivalent. Just had a look if I can quickly implement this. Will do in the coming days. I think it is a useful addition.

**Reply by jvde-github** [reply, marked as answer]
Ok, did a quick prototype, need some more testing, but looks doable without too much change

**Reply by laleft** [reply]
Hello, Jasper! Thanks for the quick response!!

I'd be happy to test it once the change is available.

Thank you very much for your time and for this awesome project!

**Comment by jvde-github** [comment]
Just uploaded the work from last night. If you want to send JSON with full decoding as a TCP client:
```
AIS-catcher -P localhost 32000 JSON_FULL on
````
with `netcat -l -p 32000` you will see full JSON on the receiving end. Notice that `JSON on` will only send over the shortened version, which is the NMEA plus the meta data.

Give it a try :-)
